### ISSUE TITLE
Training on a single GPU

### ISSUE THREAD

**Issue by tarunn2799** [salesforce/ALBEF#11]
Hi, kudos on your work, and thank you for releasing the code. I'm looking to use ALBEF to train on my custom data of about 50k-100k image-text pairs, and I had a few doubts.  I'm fairly new to this. 

- Can you tell me how I should go about changing the pretraining command to support training on a single gpu?
- What is the minimum batch size ALBEF must be trained on to get practical results (since contrastive loss is involved)
- How many epochs is recommended when trained on a single gpu, in order to reach pretraining convergence?
- I noticed in the sample pretrain config file that `train_files` is a list containing multiple jsons of different datasets. For my single dataset json, should the value for this key be a list containing path(str) to json or just the path as a string itself? 

I'm fairly new to this, and I appreciate the help. Thank you

**Comment by LiJunnan1992** [comment]
Thanks for your interest! Here are my answers:

- You can just set --nproc_per_node=1 if you are using 1 gpu
- The contrastive loss uses a queue (moco style) instead of in-batch negatives, so the batch size will not drastically affect the contrastive learning performance.
- It is hard for me to recommend the optimal batch size & epoch because the hyper-parameters are usually dataset dependant. 
- If you have only one json, you can just set train_files as ["your_json_file"]

**Comment by tarunn2799** [comment]
This is very helpful. Thank you. Sorry about the confusion regarding hparams, I was just trying to get a rule of thumb around the practices you followed for benchmark datasets like Flickr30K, etc. on the paper in order to get good performance. 

This is similar to how on the SimCLR paper they mentioned that training for 500+ epochs on datasets like CIFAR/Imagenet proved better results upon linear evaluation. Sorry if this still sounds vague to you 😅

**Comment by tarunn2799** [comment]
Also, when I tried to run a sample run of the code, I'm getting a CUDA OOM error despite having the hardware to support it. I saw #6 that said it was because of the captions being too long/ the captions containing non-english words. Is there a workaround for this? How do I bypass this error? 
cc : @LiJunnan1992

**Comment by LiJunnan1992** [comment]
You can either try to decrease the batch size, or decrease the max_length during tokenization which is currently set as 30.
